core/src/test/java/com/linecorp/armeria/server/websocket/WebSocketServiceHandshakeTest.java

```diff
@@ -102,7 +102,9 @@ void http1Handshake(SessionProtocol sessionProtocol, boolean useThreadRescheduli
                                           .build();
         final RequestHeadersBuilder headersBuilder =
                 RequestHeaders.builder(HttpMethod.GET, "/chat")
-                              .add(HttpHeaderNames.CONNECTION, HttpHeaderValues.UPGRADE.toString());
+                              // It works even if the header contains multiple values
```

Comment:
Question) When I checked, it seems like this test passes with the main branch without the changes in this PR.

I'm curious, were you able to verify the same bug with the HEAD of the main branch? It seems a little odd since `ArmeriaHttpUtil#maybeWebSocketUpgrade` is only called for trailers in the server-side which doesn't seem to make sense to me.

Reply:
Thanks, @jrhee17. In toArmeria released in version 1.28.0, I found that the purgeHttp1OnlyHeaders process was dropping useless additional headers, and that is what I intended to. I'll do it again on my server and figure out what was the issue.